content/general/push/activate-subscribe.textile

```diff
@@ -44,31 +44,37 @@ Before you can activate your push device or receive push notifications, you must
+      // TODO Handle the message delivered to the device.
+  }
+
+  @Override
+  public void onNewToken(@NonNull String s) {
+    super.onNewToken(s);
+    //Store token in Ably
+    ActivationContext.getActivationContext(this).onNewRegistrationToken(RegistrationToken.Type.FCM, s);
+  }
 }
 ```
 
 h3. Install Ably for Apple Push Notifications on iOS
 
 If you haven't yet, you should first "set up APNs":https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/APNSOverview.html in your app.
 
-After setting up APNs, you should have a couple of methods in your @UIApplicationDelegate@: <span lang="swift">"@application(_:didRegisterForRemoteNotificationsWithDeviceToken:)@":https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622958-application</span><span lang="objc">"@application:didRegisterForRemoteNotificationsWithDeviceToken:@":https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622958-application?language=objc</span> and <span lang="swift">"@application(_:didFailToRegisterForRemoteNotificationsWithError:)@":https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622962-application</span><span lang="objc">"@application:​did​Fail​To​Register​For​Remote​Notifications​With​Error:​@":https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622962-application?language=objc</span>. @ARTPush@ has two corresponding methods that you should call from yours, passing to them also an @ARTRest@ or @ARTRealtime@ instance, configured with the authentication setup and other options you need.
+In @UIApplicationDelegate#didFinishLaunchingWithOptions@, call @UIApplication.shared.registerForRemoteNotifications@. An example showing how to do this is available in the "iOS Push Notifications tutorial":https://ably.com/tutorials/ios-push-notifications#step6-register-device-for-push. You also need to override 2 methods in @UIApplicationDelegate@: <span lang="swift">"@application(_:didRegisterForRemoteNotificationsWithDeviceToken:)@":https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622958-application</span><span lang="objc">"@application:didRegisterForRemoteNotificationsWithDeviceToken:@":https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622958-application?language=objc</span> and <span lang="swift">"@application(_:didFailToRegisterForRemoteNotificationsWithError:)@":https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622962-application</span><span lang="objc">"@application:​did​Fail​To​Register​For​Remote​Notifications​With​Error:​@":https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622962-application?language=objc</span>. 
+
+@ARTPush@ has two corresponding methods that you should call from yours, passing to them also an @ARTRest@ or @ARTRealtime@ instance, configured with the authentication setup and other options you need.
```

Comment:
"passing to them also" -> "also passing to them an..." or, better still, "call from yours, passing an @ARTTest@..."

Reply:
Reworded in 17b780fdc3fb6bcc99417898416a2b1532374e42. I didn't write that initial sentence, but in this commit I have improved it.